dpc/src/state/mod.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+pub mod input;
+pub use input::*;
+
+pub mod state;
+pub use state::*;
+
+pub mod output;
+pub use output::*;
+
+// #[cfg(test)]
```

Comment:
Just to confirm: so there is a plan to add some tests?

Reply:
Yes, the plan in my next PR is to add these. The approach will be as follows:

Replace current approach to preparing inputs and outputs with `State` builder, and use the old approach as the sanity check in the unit tests against the new builder